JOSESwift/Sources/JWEHeader.swift

```diff
@@ -95,55 +101,105 @@ extension JWEHeader: CommonHeaderParameterSpace {
     /// The JWK Set URL which refers to a resource for a set of JSON-encoded public keys,
     /// one of which corresponds to the key used to encrypt the JWE.
     public var jku: URL? {
-        return parameters["jku"] as? URL
+        set {
+            parameters["jku"] = newValue
+        }
+        get {
+            return parameters["jku"] as? URL
+        }
     }
 
     /// The JSON Web key corresponding to the key used to encrypt the JWE.
     public var jwk: String? {
-        return parameters["jwk"] as? String
+        set {
+            parameters["jwk"] = newValue
+        }
+        get {
+            return parameters["jwk"] as? String
+        }
     }
 
     /// The Key ID indicates the key which was used to encrypt the JWE.
     public var kid: String? {
-        return parameters["kid"] as? String
+        set {
+            parameters["kid"] = newValue
+        }
+        get {
+            return parameters["kid"] as? String
+        }
     }
 
     /// The X.509 URL that referes to a resource for the X.509 public key certificate
     /// or certificate chain corresponding to the key used to encrypt the JWE.
     public var x5u: URL? {
-        return parameters["x5u"] as? URL
+        set {
+            parameters["x5u"] = newValue
+        }
+        get {
+            return parameters["x5u"] as? URL
+        }
     }
 
     /// The X.509 certificate chain contains the X.509 public key certificate or
     /// certificate chain corresponding to the key used to encrypt the JWE.
     public var x5c: [String: Any]? {
```

Comment:
Could you please change this parameter to `[String]` as per  [RFC-7515](https://tools.ietf.org/html/rfc7515#section-4.1.6)? Then we'll only have `[String]` and `String` parameters and the encoding to JSON data will work just fine. 😄